rxkeystore/src/androidTest/java/com/nexenio/rxkeystore/provider/asymmetric/rsa/RxRSACryptoProviderTest.java

```diff
@@ -5,22 +5,44 @@
 import com.nexenio.rxkeystore.provider.asymmetric.RxAsymmetricCryptoProvider;
 
 import org.junit.Before;
+import org.junit.Ignore;
 import org.junit.Test;
 
+import java.nio.charset.StandardCharsets;
+
 import androidx.annotation.NonNull;
 
 public class RxRSACryptoProviderTest extends BaseAsymmetricCryptoProviderTest {
 
     @Before
-    public void setUp() {
-        super.setUp();
+    @Override
+    public void setUpBeforeEachTest() {
+        super.setUpBeforeEachTest();
     }
 
     @Override
     protected RxAsymmetricCryptoProvider createAsymmetricCryptoProvider(@NonNull RxKeyStore keyStore) {
         return new RxRSACryptoProvider(keyStore);
     }
 
+    @Ignore("DHKeyAgreement requires DHPrivateKey")
+    @Test
+    @Override
+    public void generateSecretKey_matchingKeyPairs_sameSecretKey() {
+        super.generateSecretKey_matchingKeyPairs_sameSecretKey();
+    }
+
+    @Test
+    public void encrypt_invalidDataLength_emitsError() {
+        byte[] unencryptedBytes = LOREM_IPSUM_LONG.getBytes(StandardCharsets.UTF_8);
```

Comment:
In another test you changed `LOREM_IPSUM_LONG.getBytes(StandardCharsets.UTF_8)` to `new byte[]{}`.

Reply:
This is actually a different test (also different name), that tests input data that is longer than the block size.